Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanViewModel.kt

```diff
@@ -1,76 +1,41 @@
 package de.rki.coronawarnapp.ui.submission.qrcode.scan
 
-import dagger.assisted.Assisted
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
 import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQrCodeValidator
 import de.rki.coronawarnapp.coronatest.qrcode.InvalidQRCodeException
-import de.rki.coronawarnapp.submission.SubmissionRepository
-import de.rki.coronawarnapp.submission.TestRegistrationStateProcessor
 import de.rki.coronawarnapp.ui.submission.viewmodel.SubmissionNavigationEvents
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
 import de.rki.coronawarnapp.util.permission.CameraSettings
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
-import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
-import kotlinx.coroutines.flow.first
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
 import timber.log.Timber
 
 class SubmissionQRCodeScanViewModel @AssistedInject constructor(
     dispatcherProvider: DispatcherProvider,
-    @Assisted private val isConsentGiven: Boolean,
     private val cameraSettings: CameraSettings,
-    private val registrationStateProcessor: TestRegistrationStateProcessor,
-    private val submissionRepository: SubmissionRepository,
     private val qrCodeValidator: CoronaTestQrCodeValidator,
 ) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
 
     val events = SingleLiveEvent<SubmissionNavigationEvents>()
     val qrCodeErrorEvent = SingleLiveEvent<Exception>()
-    val registrationState = registrationStateProcessor.state.asLiveData2()
 
     fun registerCoronaTest(rawResult: String) = launch {
         try {
-            val ctQrCode = qrCodeValidator.validate(rawResult)
-
-            val coronaTest = submissionRepository.testForType(ctQrCode.type).first()
-            when {
-                coronaTest != null -> events.postValue(
-                    SubmissionNavigationEvents.NavigateToDuplicateWarningFragment(
-                        coronaTestQRCode = ctQrCode,
-                        consentGiven = isConsentGiven
-                    )
-                )
-
-                else -> if (!ctQrCode.isDccSupportedByPoc) {
-                    registrationStateProcessor.startRegistration(
-                        request = ctQrCode,
-                        isSubmissionConsentGiven = isConsentGiven,
-                        allowReplacement = false
-                    )
-                } else {
-                    events.postValue(
-                        SubmissionNavigationEvents.NavigateToRequestDccFragment(ctQrCode, isConsentGiven)
-                    )
-                }
-            }
+            qrCodeValidator.validate(rawResult)
+            events.postValue(SubmissionNavigationEvents.NavigateToConsent(qrCode = rawResult))
```

Comment:
this is scan fragment ,the test  registration should not be done here at all we should register after the user has given the consent and OS consent too ,  right ?

So I assume that registration should be done after the OS consent.
and we check for:
1- Duplicate test 
2- Request Dcc
3- Register and navigate to the right Test result screen

Reply:
Yes. That's why I've removed all registration related logic from the scanner.

Reply:
but where did you add it ? which screen now is registering the test ?